src/main/kotlin/uk/gov/dluhc/printapi/rest/TemporaryCertificateController.kt

```diff
@@ -1,18 +1,37 @@
 package uk.gov.dluhc.printapi.rest
 
 import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.Authentication
+import org.springframework.web.bind.WebDataBinder
 import org.springframework.web.bind.annotation.CrossOrigin
+import org.springframework.web.bind.annotation.InitBinder
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
+import uk.gov.dluhc.printapi.models.GenerateTemporaryCertificateRequest
+import uk.gov.dluhc.printapi.validator.GenerateTemporaryCertificateRequestValidator
+import javax.validation.Valid
 
 @RestController
 @CrossOrigin
-class TemporaryCertificateController {
+class TemporaryCertificateController(
+    private val generateTemporaryCertificateRequestValidator: GenerateTemporaryCertificateRequestValidator
+) {
+
+    @InitBinder("generateTemporaryCertificateRequest")
+    fun initGenerateTemporaryCertificateRequestValidator(binder: WebDataBinder) {
+        binder.addValidators(generateTemporaryCertificateRequestValidator)
```

Comment:
We need to bind a custom validator to the `generateTemporaryCertificateRequest` request parameter. This is over and above the standard JSR-303 validation that is already implemented by virtue of the fact that the openAPI codegen using standard validation annotations.
EG: this in the openAPI spec:
```
        gssCode:
          type: string
          example: E12345678
          minLength: 9
          maxLength: 9
          description: GSS code of the ERO responsible for the application
```
Becomes this in the generated model class:
```
    @get:Size(min=9,max=9)
    @Schema(example = "E12345678", required = true, description = "GSS code of the ERO responsible for the application")
    @field:JsonProperty("gssCode", required = true) val gssCode: kotlin.String,
```

We need to add a custom validator to validate the `validOnDate` field in the request. The business requirements are that the date cannot be in the past, and cannot be greater than 30 days in the future. It is not possible to express this logic in openAPI specs, hence we need the custom validator.

(to be fair it's a different type of validation. JSR-303 validation is static _format_ validation . The validation required for `validOnDate` is a business rule rather than format validation)

Reply:
I would expected the `onDate` validations to be in the service layer as another business valdiation.

Reply:
Agreed. As discussed I'm going to remove this validator class and anything related to it, and re-introduce it in my next PR where I introduce the service layer 👍